imagecraft/services/pack.py

```diff
@@ -43,15 +49,63 @@ def __init__(
         self._build_plan = build_plan
 
     @override
-    def pack(self, prime_dir: pathlib.Path, dest: pathlib.Path) -> list[pathlib.Path]:  # noqa: ARG002
+    def pack(self, prime_dir: Path, dest: Path) -> list[Path]:  # noqa: ARG002
         """Pack the image.
 
         :param prime_dir: Directory path to the prime directory.
         :param dest: Directory into which to write the package(s).
         :returns: A list of paths to created packages.
         """
+        # Pydantic has already validated that there is only a single volume before now
+        project = cast(Project, self._project)
+        if len(project.volumes) != 1:
+            raise AssertionError("This code can only handle one volume")
+        volume_name, volume = next(iter(project.volumes.items()))
+        disk_image_file = dest / (volume_name + os.extsep + "img")
+
+        # Create empty image
+        gptutil.create_empty_gpt_image(
+            imagepath=disk_image_file,
+            sector_size=SECTOR_SIZE,
+            layout=volume,
+        )
+
+        # Create partition images with filesystems.  These are always recreated, but we
+        # may want to revisit that once this is solved:
+        # https://github.com/canonical/craft-parts/issues/665
+        project_dirs = self._services.lifecycle.project_info.dirs
+        with tempfile.TemporaryDirectory() as partition_dir:
```

Comment:
This could be a problem in case of very large images and limited `/tmp` capacity but we can address that later if needed.

Reply:
Good catch. This *will* be a problem even for moderate images since `/tmp` is now rather small by default.